### DISCUSSION TITLE
Caching plugins between execution on docker images (Through gitlab ci)

### DISCUSSION THREAD

**Discussion by jacobdonenfeld** [terraform-linters/tflint#1518] [Q&A]
Looking to cache the plugin directory between execution on gitlab's CI. Currently running tflint as a pre-commit hook, and running pre-commit on the job. Here is what I am doing, and where I am stuck:

If a previous run's job cache is available, set the tflint plugin path to the known cached plugin directory:
`if [ -d ".tflint.d/plugins" ]; then export TFLINT_PLUGIN_DIR=./.tflint.d/plugins; fi`

Initialize the plugins, in case they are out of date, or there wasn't a cache:
`tflint --init`

Slightly unrelated:
- install pre-commit hooks:
`pre-commit install-hooks`
- run pre-commit, in turn, run tflint:
`pre-commit run --all-files || status=$?`

**The issue I am running into** is on running `tflint --init`, the command is running into some missing directory issue:
```
Installing `aws` plugin...
Failed to install a plugin; Failed to download checksums.txt: open /tmp/tflint-download-temp-file-442339605: no such file or directory
```
Haven't been able to solve this. Would like to avoid defining the plugins in two locations or any complicated scripting to get this to work. 

Thanks in advanced!

**Comment by wata727** [comment]
According to the error message it seems to fail to create the tmp file, but I'm not sure why that happens.
https://github.com/terraform-linters/tflint/blob/a09464b192504e5182bf313a555bf915a1085425/plugin/install.go#L183
